docs/modules/examples/pages/knowledge-graph.adoc

```diff
@@ -0,0 +1,288 @@
+* OpenAI API key
+
+== Environment
+
+. Install dependencies:
++
+[source,bash]
+----
+pip install ragstack-ai-knowledge-graph python-dotenv
+----
++
+. Create a `.env` file and store the necessary credentials.
++
+[source,bash]
+----
+OPENAI_API_KEY="sk-..."
+ASTRA_DB_DATABASE_ID="670d40c2-80f9-4cb0-8c74-d524dd6944d1"
+ASTRA_DB_APPLICATION_TOKEN="AstraCS:..."
+ASTRA_DB_KEYSPACE="default_keyspace"
+----
+
+If you're running the notebook in Colab, run the cell using `getpass` to set the necessary environment variables.
+
+== Create a graph store in Astra
+
+. Import the necessary libraries and load the variables from your `.env` files.
++
+[source,python]
+----
+import dotenv
+import cassio
+from ragstack_knowledge_graph.cassandra_graph_store import CassandraGraphStore
+from langchain_experimental.graph_transformers import LLMGraphTransformer
+from langchain_openai import ChatOpenAI
+from langchain_core.documents import Document
+from ragstack_knowledge_graph.render import render_graph_documents
+from ragstack_knowledge_graph.traverse import Node
+from ragstack_knowledge_graph import extract_entities
+from operator import itemgetter
+from langchain_core.runnables import RunnableLambda, RunnablePassthrough
+from langchain_core.prompts import ChatPromptTemplate
+
+dotenv.load_dotenv()
+----
++
+. Initialize a connection to AstraDB with the Cass-IO library.
++
+[source,python]
+----
+import cassio
+cassio.init(auto=True)
+----
++
+. Create graph store.
++
+[source,python]
+----
+from knowledge_graph.cassandra_graph_store import CassandraGraphStore
+graph_store = CassandraGraphStore()
+----
+
+== Extract a knowledge graph from your data
+
+. Extract a knowledge graph with LLMGraphTransformer, and render it to Astra with GraphViz.
++
+[source,python]
+----
+llm = ChatOpenAI(temperature=0, model_name="gpt-4")
+
+llm_transformer = LLMGraphTransformer(llm=llm)
+
+text = """
+Marie Curie, was a Polish and naturalised-French physicist and chemist who conducted pioneering research on radioactivity.
+She was the first woman to win a Nobel Prize, the first person to win a Nobel Prize twice, and the only person to win a Nobel Prize in two scientific fields.
+Her husband, Pierre Curie, was a co-winner of her first Nobel Prize, making them the first-ever married couple to win the Nobel Prize and launching the Curie family legacy of five Nobel Prizes.
+She was, in 1906, the first woman to become a professor at the University of Paris.
+"""
+documents = [Document(page_content=text)]
+graph_documents = llm_transformer.convert_to_graph_documents(documents)
+print(f"Nodes:{graph_documents[0].nodes}")
+print(f"Relationships:{graph_documents[0].relationships}")
+----
++
+. Render the extracted graph to GraphViz and save the extracted graph documents to the AstraDB graph store.
++
+[source,python]
+----
+render_graph_documents(graph_documents)
+graph_store.add_graph_documents(graph_documents)
+----
+
+=== Optional: Predefine relationships between entities
```

Comment:
Can probably drop this -- the knowledge schema stuff is more robust, and we can add a notebook / docs on that at some point.